File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/tng_do8"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your TNG DO8.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	do8, _ := tng_do8.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	do8.SetValues(0, [8]bool{true, false, true, false, true, false, true, false})

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{do8_tng::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your TNG DO8.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let do8 = Do8Tng::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    do8.set_values(0, [true, false, true, false, true, false, true, false]).recv()?;

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
